doc/source/overview.rst

```diff
@@ -60,6 +60,64 @@ objects, have a look a :doc:`dev_guide/satpy_internals`.
     XArray includes other object types called "Datasets". These are different
     from the "Datasets" mentioned in Satpy.
 
+Data chunks
+-----------
+
+The usage of dask as ground layer for Satpy's operation means that the
+underlying data is chunked, that is cut in smaller pieces that can then be
```

Comment:
I'm not sure if there needs to be a semi-colon somewhere but a comma after this seems better...maybe?

```suggestion
underlying data is chunked, that is, cut in smaller pieces that can then be
```